tests/saitests/sai_qos_tests.py

```diff
@@ -1347,6 +1347,149 @@ def runTest(self):
+            for i in range(len(self.src_port_ids)):
+                dscp = self.dscps[i]
+                pg = self.pgs[i]
+                queue = self.queues[i]
+                src_port_id = self.src_port_ids[i]
+                dst_port_id = self.dst_port_ids[i]
+                src_port_mac = self.src_port_macs[i]
+                dst_port_mac = self.dst_port_macs[i]
+                src_port_ip = self.src_port_ips[i]
+                dst_port_ip = self.dst_port_ips[i]
+                pkt_count = self.pkt_counts[i]
+
+                tos = (dscp << 2) | self.ecn
+                ttl = 64
+                pkt = simple_tcp_packet(pktlen=self.packet_size,
+                                        eth_dst=self.router_mac if self.router_mac != '' else dst_port_mac,
+                                        eth_src=src_port_mac,
+                                        ip_src=src_port_ip,
+                                        ip_dst=dst_port_ip,
+                                        ip_tos=tos,
+                                        ip_ttl=ttl)
+
+                if i == len(self.src_port_ids) - 1:
+                    # Verify XOFF has not been triggered on final port before sending traffic
+                    print >> sys.stderr, "Verifying XOFF hasn't been triggered yet on final iteration"
+                    sys.stderr.flush()
+                    time.sleep(8)
+                    recv_counters = sai_thrift_read_port_counters(self.client, port_list[src_port_id])[0]
+                    xoff_txd = recv_counters[self.pg_cntr_indices[i]] - recv_counters_bases[i][self.pg_cntr_indices[i]]
+                    assert xoff_txd == 0, "XOFF triggered too early on final iteration, XOFF count is %d" % xoff_txd
+
+                # Send requested number of packets
+                print >> sys.stderr, "Sending %d packets for dscp=%d, pg=%d, src_port_id=%d, dst_port_id=%d" % (pkt_count, dscp, pg, src_port_id, dst_port_id)
+                sys.stderr.flush()
+                if 'cisco-8000' in self.asic_type:
+                    assert(fill_leakout_plus_one(self, src_port_id, dst_port_id, pkt, queue, self.asic_type))
+                    pkt_count -= 1 # leakout adds 1 packet, subtract from current iteration
+
+                send_packet(self, src_port_id, pkt, pkt_count)
+
+                if i == len(self.src_port_ids) - 1:
+                    # Verify XOFF has now been triggered on final port
+                    print >> sys.stderr, "Verifying XOFF has now been triggered on final iteration"
+                    sys.stderr.flush()
+                    time.sleep(8)
+                    recv_counters = sai_thrift_read_port_counters(self.client, port_list[src_port_id])[0]
+                    xoff_txd = recv_counters[self.pg_cntr_indices[i]] - recv_counters_bases[i][self.pg_cntr_indices[i]]
+                    assert xoff_txd > 0, "Failed to trigger XOFF on final iteration"
+
+            # Verify no ingress/egress drops for all ports
```

Comment:
In the original case testQosSaiHeadroomPoolSize, after checking the xoff trigger, it will also check the packet will be dropped once headroom is full, any reason there isn't such implementation in the new case?